### ISSUE TITLE
ANNOUNCEMENT: Please help testing 8.0.0-beta.9

### ISSUE THREAD

**Issue by pubkey** [pubkey/rxdb#781]
Hello everybody.
Today I released `rxdb@8.0.0-beta.9` with improvements and changes.

The focus of the new version lays on **better defaults** and **better performance**.

Please help testing the new version by installing it via `npm i rxdb@8.0.0-beta.9` and report any bugs you may find.

What has changed?:
- [Changelog](https://github.com/pubkey/rxdb/blob/8.0.0/CHANGELOG.md)
- [Release blogpost](https://github.com/pubkey/rxdb/blob/8.0.0/orga/releases/8.0.0.md)

Every change in the current beta is still changeable and you are invited to discussion.
If you have any other proposals for breaking changes, please speak now or hold your peace until the next major release.

EDIT: Released `8.0.0-beta.2`
EDIT: Released `8.0.0-beta.3`
EDIT: Released `8.0.0-beta.4`
EDIT: Released `8.0.0-beta.5`
EDIT: Released `8.0.0-beta.6`
EDIT: Released `8.0.0-beta.7`
EDIT: Released `8.0.0-beta.8`
EDIT: Released `8.0.0-beta.9`

**Comment by pubkey** [comment]
@ngohuunam Oh yes, I forgot to mention. Please update vuejs to the latest version to fix this problem.

**Comment by pubkey** [comment]
@ngohuunam can you reproduce the issue in the vue-example?

**Comment by pubkey** [comment]
Released `8.0.0-beta.2`
- Fixed broken broadcast-channel on OSX
- Fixed problem with `rxjs@6.3.1`

**Comment by erguotou520** [comment]
Seems not work with hooks.
It throws `Since version 8.0.0 RxDocument.set() can only be called on temporary RxDocuments`.
This throws in `preSave` hook, and the hook function looks like
```js
export function preSave (doc) {
  const now = +new Date()
  if (!doc.__skip) {
    doc.updateAt = now
  }
  if (doc.removed && !doc.removeAt) {
    doc.removeAt = now
  } else if (!doc.removed && doc.removeAt) {
    delete doc.removeAt
  } else if (!doc.enabled && !doc.disableAt) {
    doc.disableAt = now
  } else if (doc.enabled && doc.disableAt) {
    delete doc.disableAt
  }
}
```

**Comment by erguotou520** [comment]
And a lot of schema not match error when migrate to `8.0.0-beta.2` 😢

**Comment by ngohuunam** [comment]
@pubkey here the test link [https://github.com/ngohuunam/test0409](url)
- run npm update -g to update lastest vue-cli
- create new Vue with Router, Vuex, ServiceWorker
- install rxdb@8.0.0-beta.2, rxjs, pouchdb-adapter-http, pouchdb-adapter-idb, pouchdb-server, random-int, concurrently
- copy folder components, database from your vue example
- create vuex plugin to init rxdb and store heroes rxcollection, observable$ to vuex
- re write some component code to fit with vuex
- error was thrown if enable Vue devtool extension on chrome or firefox, no error if disable Vue devtool
I quite sure it's becaues I store RxDocument to Vuex store

**Comment by erguotou520** [comment]
Yes, I think the `RxDocument in vuex/data` operation caused my migration failed probability.

**Comment by pubkey** [comment]
Released 8.0.0-beta.3 With refactored middleware-hooks: https://github.com/pubkey/rxdb/blob/8.0.0/orga/releases/8.0.0.md#middleware-hooks-contain-plain-json-as-first-parameter-and-rxdocument-as-second

**Comment by ngohuunam** [comment]
@pubkey thanks for your work.
I know it is bad if store self-mutating object into vuex, but with my small project, all datas use as datatable, with rxdocument as a table row, any data changed will update to the view and I do not need more code to handle data changed.

**Comment by pubkey** [comment]
Released 8.0.0-beta.4 with a fix for electron and optimised collection-creation-time.

**Comment by pubkey** [comment]
@ngohuunam Yes I can understand your use case. Also RxDB should not tell you what to put into the store and what not. The problem is that I spend 4 hours trying to fix this, but could not. Also I think this is an edge case. A PR with a fix would be nice but I will not spend more time for this.
I belived that vuex is doing a clone on the `RxDocument` somewhere which will then loose the prototype with the getters.

**Comment by ngohuunam** [comment]
@pubkey yeah, I will try some investigate, hope some luck and thanks for your time

**Comment by ngohuunam** [comment]
@pubkey after investigated, I think this is the problem. Would you take a look? this print out from hero-list component code, not from vuex, 
```javascript
import * as DB from '../database/Database'

mounted: async function() {
    const db = await DB.get()
    const docs = await db.heroes.find().exec()
    console.log(docs)
  },
```
![error-rx-document](https://user-images.githubusercontent.com/3610240/45211593-e7f52a80-b2bc-11e8-9a9a-d14822185a16.png)
Anyway to use rxdb v7.x with pouchdb 7?

**Comment by pubkey** [comment]
Released `8.0.0-beta.5` with bugfix and reduced build-size.

**Comment by pubkey** [comment]
@ngohuunam what your console-screenshot shows is intended. `_getDocumentPrototype` returns a prototype only which is used together with `getDocumentConstructor` to create the 'class' for documents of this collection.

**Comment by ngohuunam** [comment]
@pubkey I try to change `__defineGetter__` and` __defineSetter__` to `defineProperty` in rx-document and there is no error thrown.
The vue dev tool take a snapshot of vue component data or vuex state from the app in dev mod by stringify them, when it call native `JSON.stringify` it throw `TypeError: Converting circular structure to JSON` and vue dev tool turn to another way called `CircularJSON` that make a deep loop through the datas of object, when the loop reach the property `collection._getDocumentPrototype.name` it throw error (the sreenshot in my first report), than the vue dev tool doesn't show data.

**Comment by pubkey** [comment]
@ngohuunam ok that gives a hint. Do you which tool vuejs uses for the cirular clone?

**Comment by ngohuunam** [comment]
@pubkey as you said, this is an edge case, and we do not need to spend time for it and I find the way the by pass this error.

**Comment by pubkey** [comment]
Released `8.0.0-beta.6` with better typings (BREAKING) and tutorials for rxdb+typescript and the server-plugin.

**Comment by pubkey** [comment]
Released `8.0.0-beta.7` with CORS for the server-plugin and this-scope-binding on the collection-hooks.

**Comment by pubkey** [comment]
Released `8.0.0-beta.8`

**Comment by pubkey** [comment]
If someone finds the time, it would be nice if you can fix the [react-native-example](https://github.com/pubkey/rxdb/tree/8.0.0/examples/react-native), there is something broken with the tests, caused by the dependencies.

`8.0.0` will be released if there are not more bug-reports in 2 ongoing workdays.

**Comment by pubkey** [comment]
Released `8.0.0-beta.9` where `disabledKeyCompression` is renamed to `keyCompression`

**Comment by pubkey** [comment]
`8.0.0` is now released. For future bugs, open an issue as normal.
